pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -1432,6 +1439,110 @@ func SyncVMSnapshotTreeStatus(
 	return updateRootSnapshots(vmCtx, k8sClient)
 }
 
+// updateSnapshotTreeChildrenStatus updates Status.Children for
+// every snapshot in the VM's snapshot tree.
+func updateSnapshotTreeChildrenStatus(
+	vmCtx pkgctx.VirtualMachineContext,
+	k8sClient ctrlclient.Client) error {
+	mo := vmCtx.MoVM
+	if mo.Snapshot == nil || len(mo.Snapshot.RootSnapshotList) == 0 {
+		return nil
+	}
+
+	for _, root := range mo.Snapshot.RootSnapshotList {
+		if _, err := updateSnapshotChildrenStatus(vmCtx, k8sClient, root); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// updateSnapshotChildrenStatus returns the LocalObjectRef
+// for the current snapshot (if CR exists),
+// and ensures its Status.Children is up-to-date.
+func updateSnapshotChildrenStatus(
+	vmCtx pkgctx.VirtualMachineContext,
+	k8sClient ctrlclient.Client,
+	node vimtypes.VirtualMachineSnapshotTree,
+) (*common.LocalObjectRef, error) {
+	curSnapshot, err := getSnapshotCR(vmCtx, k8sClient, node.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	if curSnapshot == nil {
+		// TODO (Lubron): This is for external snapshots.
+		vmCtx.Logger.V(4).Info("Snapshot custom resource not found, skipping",
+			"snapshotName", node.Name)
+		return nil, nil
+	}
+
+	// Recurse into children and collect their refs (only if their CRs exist).
+	children := make([]common.LocalObjectRef, 0, len(node.ChildSnapshotList))
+	for i := range node.ChildSnapshotList {
+		child := node.ChildSnapshotList[i]
+		ref, err := updateSnapshotChildrenStatus(vmCtx, k8sClient, child)
+		if err != nil {
+			return nil, err
+		}
+		// TODO (Lubron): If ref is nil, it means the snapshot custom resource is not found
+		// and it's an external snapshot.
+		if ref != nil {
+			children = append(children, *ref)
+		}
+	}
+
+	// Update Status.Children if changed.
+	if !reflect.DeepEqual(curSnapshot.Status.Children, children) {
```

Comment:
children is a slice, so it may not be ordered and thereforce, the deepequal may fail?